src/Core/OrganizationFeatures/AuthorizationHandlers/OrganizationAuthHandler.cs

```diff
@@ -0,0 +1,41 @@
+﻿using Bit.Core.Context;
+using Bit.Core.Enums;
+using Microsoft.AspNetCore.Authorization;
+
+namespace Bit.Core.OrganizationFeatures.AuthorizationHandlers;
+
+class OrganizationAuthHandler : AuthorizationHandler<OrganizationOperationRequirement, CurrentContentOrganization>
+{
+    protected override async Task HandleRequirementAsync(AuthorizationHandlerContext context,
+        OrganizationOperationRequirement requirement,
+        CurrentContentOrganization resource)
+    {
+        if (requirement == OrganizationOperations.ReadAllGroups)
+        {
+            await ReadAllGroupsAsync(context, requirement, resource);
+            return;
+        }
+    }
+
+    private async Task ReadAllGroupsAsync(AuthorizationHandlerContext context,
+        OrganizationOperationRequirement requirement,
+        CurrentContentOrganization resource)
+    {
+        // TODO: providers need to be included in the claims
+
+        var canAccess = resource.Type == OrganizationUserType.Owner ||
+            resource.Type == OrganizationUserType.Admin ||
+            resource.Type == OrganizationUserType.Manager ||
+            (resource.Permissions?.ManageGroups ?? false) ||
+            (resource.Permissions?.EditAssignedCollections ?? false) ||
+            (resource.Permissions?.DeleteAssignedCollections ?? false) ||
+            (resource.Permissions?.CreateNewCollections ?? false) ||
+            (resource.Permissions?.EditAnyCollection ?? false) ||
+            (resource.Permissions?.DeleteAnyCollection ?? false);
```

Comment:
This long chain of `?.* ?? false` makes me think we should have a default permissions set somewhere and during construction if `Permissions` is null we can set it to that and simplify a lot of these checks.

Reply:
Good point - I think https://github.com/bitwarden/server/pull/2845/commits/22fa5b8170632fbc78b7f24f7ec5e81b95e761ef should do the trick?